File: backend/src/main/java/org/ilisi/backend/model/Experience.java
package org.ilisi.backend.model;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.YearMonth;

@Entity
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Experience {
    @Id
    @GeneratedValue(strategy = GenerationType.UUID)
    @Column(updatable = false, nullable = false)
    private String id;

    private String title;
    private String description;
    private YearMonth startAt;
    private YearMonth endAt;
    private String location;

    @ManyToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "institu_id", nullable = false)
    private Institut institut;
}

File: frontend/src/app/models/experience.ts
class Experience {
    id?: string;
    title: string;
    description: string;
    startAt: string;
    endAt: string;
    institut?: any;

    constructor(
        id: string,
        title: string,
        description: string,
        startAt: string,
        endAt: string,
        institut: any
    ) {
        this.id = id;
        this.title = title;
        this.description = description;
        this.startAt = startAt;
        this.endAt = endAt;
        this.institut = institut;
    }
}

export default Experience;
